pyop2/base.py

```diff
@@ -335,131 +331,109 @@ def split(self):
         else:
             return (self,)
 
-    @property
+    @cached_property
     def name(self):
         """The generated argument name."""
-        return "arg%d" % self._position
-
-    @property
-    def position(self):
-        """The position of this :class:`Arg` in the :class:`ParLoop` argument list"""
-        return self._position
-
-    @position.setter
-    def position(self, val):
-        """Set the position of this :class:`Arg` in the :class:`ParLoop` argument list"""
-        self._position = val
-
-    @property
-    def indirect_position(self):
-        """The position of the first unique occurence of this
-    indirect :class:`Arg` in the :class:`ParLoop` argument list."""
-        return self._indirect_position
-
-    @indirect_position.setter
-    def indirect_position(self, val):
-        """Set the position of the first unique occurence of this
-    indirect :class:`Arg` in the :class:`ParLoop` argument list."""
-        self._indirect_position = val
+        return "arg%d" % self.position
```

Comment:
So property access is slower than slot access, and every time we build a par_loop we set these values.  So now we just set the public slots directly, rather than going through the property slots.